modules/renter/persist.go

```diff
@@ -1,44 +1,74 @@
 package renter
 
 import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
 	"path/filepath"
+)
+
+const (
+	PersistFilename = "renter.dat"
+	PersistHeader   = "Renter Persistence"
+	PersistVersion  = "0.2"
+)
 
-	"github.com/NebulousLabs/Sia/encoding"
-	"github.com/NebulousLabs/Sia/types"
+var (
+	ErrUnrecognizedHeader  = errors.New("renter persistence file has unrecognized header")
+	ErrUnrecognizedVersion = errors.New("renter persistence file has unrecognized version")
 )
 
-// savedFiles contains the list of all the files that have been saved by the
-// renter.
-type savedFiles struct {
-	FilePieces  []FilePiece
-	Nickname    string
-	StartHeight types.BlockHeight
+// RenterPersistence is the struct that gets written to and read from disk as
+// the renter is saved and loaded.
+type RenterPersistence struct {
+	Header  string
+	Version string
+	Files   []file
 }
 
-// save puts all of the files known to the renter on disk.
+// save stores the current renter data to disk.
 func (r *Renter) save() error {
-	// create slice of savedFiles
-	savedPieces := make([]savedFiles, 0, len(r.files))
-	for nickname, file := range r.files {
-		savedPieces = append(savedPieces, savedFiles{file.pieces, nickname, file.startHeight})
+	rp := RenterPersistence{
+		Header:  PersistHeader,
+		Version: PersistVersion,
+		Files:   make([]file, 0, len(r.files)),
+	}
+	for _, file := range r.files {
+		rp.Files = append(rp.Files, *file)
+	}
+
+	persistBytes, err := json.Marshal(rp)
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(filepath.Join(r.saveDir, PersistFilename), persistBytes, 0660)
+	if err != nil {
+		return err
 	}
-	return encoding.WriteFile(filepath.Join(r.saveDir, "files.dat"), savedPieces)
+	return nil
 }
 
-// load loads all of the files from disk.
+// load fetches the saved renter data from disk.
 func (r *Renter) load() error {
-	var pieces []savedFiles
-	err := encoding.ReadFile(filepath.Join(r.saveDir, "files.dat"), &pieces)
+	persistBytes, err := ioutil.ReadFile(filepath.Join(r.saveDir, PersistFilename))
 	if err != nil {
 		return err
 	}
-	for _, piece := range pieces {
-		r.files[piece.Nickname] = File{
-			nickname:    piece.Nickname,
-			pieces:      piece.FilePieces,
-			startHeight: piece.StartHeight,
-			renter:      r,
-		}
+	var rp RenterPersistence
+	err = json.Unmarshal(persistBytes, &rp)
+	if err != nil {
+		return err
+	}
+
+	if rp.Header != PersistHeader {
+		return ErrUnrecognizedHeader
+	}
+	if rp.Version != PersistVersion {
+		return ErrUnrecognizedVersion
+	}
```

Comment:
fyi, this approach wouldn't work so well with the standard encoder. You would have to separately decode the header and version first before continuing. And everything will break anyway if we change encoding schemes. So maybe sticking with JSON is a good idea.